### PR TITLE
[ARM-5] ✨ Add security scan

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

**Review by tcarmet — CHANGES_REQUESTED** [review]
Please rebase your branch onto the branch `development/1.0` and retarget your pr to it.

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [development/*]
+  pull_request:
+    branches: [development/*]
```
Comment (tcarmet) [inline .github/workflows/codeql.yaml]: we should be adding those branches:
- development/*
- stabilization/*
- hotfix/*

File: .github/workflows/dependency-review.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+name: dependency review
+
+on:
+  pull_request:
+    branches: [development/*]
```
Comment (tcarmet) [inline .github/workflows/dependency-review.yaml]: same story with the branches as above

File: .github/workflows/codeql.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: javascript
```
Comment (tcarmet) [inline .github/workflows/codeql.yaml:22]: Can you confirm there's only javascript inside this repo?

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yaml (hunk 1 above)
Comment (gaspardmoindrot) [reply .github/workflows/codeql.yaml]: Done

File: .github/workflows/dependency-review.yaml (hunk 2 above)
Comment (gaspardmoindrot) [reply .github/workflows/dependency-review.yaml]: Done

File: .github/workflows/codeql.yaml (hunk 3 above)
Comment (gaspardmoindrot) [reply .github/workflows/codeql.yaml:22]: Yes, I checked, also checked with Github
image

**Comment by tcarmet** [comment]
@gaspardmoindrot I'll let you write `/approve` in the PR to let the bot know you have given your 👍 for it to merge.

File: .github/workflows/codeql.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+      - name: Build and analyze
+        uses: github/codeql-action/analyze@v2
+        continue-on-error: true
```
Comment (tcarmet) [inline .github/workflows/codeql.yaml]: This is not achieving what you want it to do
```suggestion
```

**Review by tcarmet — CHANGES_REQUESTED** [review]
Please ensure your commit history is accurate, I see 3 commits which can either be renamed or squashed into 1.

**Review by tcarmet — APPROVED** [review]

**Pull request merged**